packages/web/src/components/usdc-purchase-details-modal/components/PurchaseModalContent.tsx

```diff
@@ -40,25 +51,51 @@ const messages = {
   done: 'Done',
   purchaseDetails: 'Purchase Details',
   track: 'Track',
+  album: 'Album',
   visitTrack: 'Visit Track',
+  visitAlbum: 'Visit Album',
   transaction: 'Explore Transaction'
 }
 
 export const PurchaseModalContent = ({
   purchaseDetails,
   onClose
-}: ContentProps) => {
+}: {
+  purchaseDetails: USDCPurchaseDetails
+  onClose: () => void
+}) => {
+  const goToRoute = useGoToRoute()
+  const { contentType, contentId } = purchaseDetails
+  const isTrack = contentType === USDCContentPurchaseType.TRACK
   const { isEnabled: isPremiumAlbumsEnabled } = useFlag(
     FeatureFlags.PREMIUM_ALBUMS_ENABLED
   )
-  const goToRoute = useGoToRoute()
-  const { data: track } = useGetTrackById({ id: purchaseDetails.contentId })
+  const { data: currentUserId } = useGetCurrentUserId({})
+  const { data: track } = useGetTrackById(
+    { id: contentId },
+    { disabled: !isTrack }
+  )
+  const { data: album } = useGetPlaylistById(
+    {
+      playlistId: contentId,
+      currentUserId
+    },
+    { disabled: isTrack }
+  )
+  const trackArtwork = useTrackCoverArt2(contentId, SquareSizes.SIZE_150_BY_150)
+  const albumArtwork = useCollectionCoverArt2(
+    contentId,
+    SquareSizes.SIZE_150_BY_150
+  )
+
```

Comment:
I agree with your comment above @dharit-tan that this component is pretty generic as-is and should be kept as 1 component. 
Suggestion: Why dont we make most/all of the render logic agnostic by setting up all the entity data up at the top of the component.
 Removes all the ternary stuff and sets us up better if we have more purchase types in the future

e.g.: 
```tsx
const getEntityData = () => { 
  if (isTrack) {
    return {
      artwork: track.artwork,
      title: track.title,
      link: track.permalink,
      label: messages.track,
      buttonText: messages.visitTrack
    }
  }
  else if (album) {
    // ...etc
  }
}
const entity = getEntityData()

// render code
```